Rubberduck.Refactorings/RemoveParameters/RemoveParametersRefactoring.cs

```diff
@@ -111,16 +110,15 @@ private void RemoveParameters()
                 throw new NullReferenceException("Parameter is null");
             }
 
-            AdjustReferences(_model.TargetDeclaration.References, _model.TargetDeclaration);
-            AdjustSignatures();
+            var rewritingSession = _rewritingManager.CheckOutCodePaneSession();
 
-            foreach (var rewriter in _rewriters)
-            {
-                rewriter.Rewrite();
-            }
+            AdjustReferences(_model.TargetDeclaration.References, _model.TargetDeclaration, rewritingSession);
+            AdjustSignatures(rewritingSession);
+
+            rewritingSession.TryRewrite();
```

Comment:
is there a contract that refactorings have to trigger the sessions's rewrite themselves? If not, I'd prefer they worked like quickfixes do ..

Reply:
I asked about the approach to take in this PR in chat. @retailcoder opted for this version. The alternative would have required to change the `IRewriter` interface.